src/components/MatrixBuilder/index.js

```diff
@@ -0,0 +1,50 @@
+import { nanoid } from "nanoid";
+import { SettingsItem } from "../SettingsItem";
+import { CustomButton } from "../shared/CustomButton";
+import styles from "./MatrixBuilder.module.scss";
+
+export const MatrixBuilder = ({ handleSetMatrix, settings, onSetSettings }) => {
+  const handleCreateMatrix = () => {
+    let matrix = Array(settings.rows)
+      .fill()
+      .map(() =>
+        Array(settings.columns)
+          .fill()
+          .map(() => ({
+            id: nanoid(),
+            num: Math.floor(Math.random() * (999 - 100) + 100),
+          }))
+      );
+    handleSetMatrix(matrix);
+  };
+
+  const handleSetSettings = (event) => {
+    const { value, name } = event.target;
+    onSetSettings({ ...settings, [name]: parseInt(value) });
```

Comment:
just use +value. add checking for the numbers less than 0 and set just zero if true